### PR TITLE
redirect: Download Kubernetes SBOMs from dl.

### PR DESCRIPTION
**Pull request by ameukam**
Ensure we download the SBOMs and their signatures from dl.k8s.io

### CODE REVIEW

**Comment by ameukam** [comment]
/hold

cc @BenTheElder @upodroid @kubernetes/release-engineering

**Review by upodroid — APPROVED** [review]
LGTM

**Review by saschagrunert — APPROVED** [review]

File: apps/k8s-io/configmap-nginx.yaml [hunk 1]
```diff
@@ -362,8 +362,8 @@ data:
         server_name sbom.k8s.io sbom.kubernetes.io
         listen 80;
 
-        rewrite ^/(.*)?/release(\.cert|\.sig|\.sha256|\.sha512)?$ https://storage.googleapis.com/kubernetes-release/release/$1/kubernetes-release.spdx$2 redirect;
-        rewrite ^/(.*)?/source(\.cert|\.sig|\.sha256|\.sha512)?$  https://storage.googleapis.com/kubernetes-release/release/$1/kubernetes-source.spdx$2 redirect;
+        rewrite ^/(.*)?/release(\.cert|\.sig|\.sha256|\.sha512)?$ https://dl.k8s.io/release/$1/kubernetes-release.spdx$2 redirect;
+        rewrite ^/(.*)?/source(\.cert|\.sig|\.sha256|\.sha512)?$  https://dl.k8s.io/release/$1/kubernetes-source.spdx$2 redirect;
```
Comment (saschagrunert) [inline apps/k8s-io/configmap-nginx.yaml:366]: Is there any drawback for older releases?

Comment (ameukam) [reply apps/k8s-io/configmap-nginx.yaml:366]: All the older releases are currently synced to the new bucket.

```bash
http -I -F https://dl.k8s.io/release/v1.22.0/kubernetes-release.spdx.sha256
HTTP/1.1 200 OK
Accept-Ranges: bytes
Age: 0
Cache-Control: private, no-store
Connection: keep-alive
Content-Length: 64
Content-Type: application/octet-stream
Date: Wed, 11 Sep 2024 09:48:37 GMT
ETag: "f98d02098f161ce5c02040f58cf3f4d2"
Last-Modified: Fri, 17 Nov 2023 20:16:08 GMT
access-control-allow-origin: *

47eb5acc2319c773ceab650b0278348bc9411bc9fc40d8d63da4b2508ac119ba
```

We shouldn't have any drawback at least for those with provenance and sboms (k8s >= 1.22.0 IIRC)

**Pull request merged**